app/scripts/directives/serviceIntegration.js

```diff
@@ -0,0 +1,242 @@
+'use strict';
+
+(function() {
+  angular.module('openshiftConsole').component('serviceIntegration', {
+    controller: [
+      '$filter',
+      'APIService',
+      'AuthorizationService',
+      'BindingService',
+      'DataService',
+      'NotificationsService',
+      ServiceIntegration
+    ],
+    bindings: {
+      integration: '<',
+      consumerService: '<?'
+    },
+    templateUrl: 'views/directives/_service-integration.html'
+  });
+
+  function ServiceIntegration(
+                          $filter,
+                          APIService,
+                          AuthorizationService,
+                          BindingService,
+                          DataService,
+                          NotificationsService) {
+    var ctrl = this;
+
+    var deploymentPreferredVersion = APIService.getPreferredVersion('deployments');
+    var bindingPreferredVersion = APIService.getPreferredVersion('servicebindings');
+    var instancePreferredVersion = APIService.getPreferredVersion('serviceinstances');
+    var podPresetPreferredVersion = {group: "settings.k8s.io", version: "v1alpha1", resource: "podpresets"};
+
+    var isServiceInstanceReady = $filter('isServiceInstanceReady');
+    var isBindingReady = $filter("isBindingReady");
+
+    var integrationName = ctrl.integration.spec.externalMetadata.serviceName;
+
+    ctrl.$onInit = function() {
+      var context = {namespace: ctrl.consumerService.metadata.namespace};
+      // watch for the pod preset for this integration
+      DataService.watch(podPresetPreferredVersion, context, function(podPresets){
+        var data = podPresets.by("metadata.name");
+        ctrl.podPreset = _.filter(data, function(podPreset) {
+          return podPreset.metadata.name === _.get(ctrl.consumerService, 'metadata.name') + "-" + _.get(ctrl.serviceInstance, 'metadata.name');
+        }).pop();
+      });
+
+      //setup watch on servincebindings, watch for bindings consumed by this service
+      DataService.watch(bindingPreferredVersion, context, function(bindingData) {
+        var data = bindingData.by("metadata.name");
+        ctrl.binding = _.filter(data, function(binding){
+          var bindingProviderName = _.get(binding, "metadata.annotations.provider", "provider_not_found");
+          var bindingConsumerName = _.get(binding, "metadata.annotations.consumer", "consumer_not_found");
+          var consumerName = _.get(ctrl, "consumerService.metadata.labels.serviceName", "consumer_name_not_found");
```

Comment:
I'd rather just check for null / undefined instead of using something like "provider_not_found".

Reply:
We typically namespace annotations. Something like `servicecatalog.openshift.io/consumer`.